core/src/core/pmmr.rs

```diff
@@ -588,15 +590,30 @@ impl PruneList {
 	pub fn get_shift(&self, pos: u64) -> Option<u64> {
 		// get the position where the node at pos would fit in the pruned list, if
 		// it's already pruned, nothing to skip
-		match self.pruned_pos(pos) {
+
+		let pruned_idx = self.next_pruned_idx(pos);
+		let next_idx = self.pruned_nodes.binary_search(&pos).map(|x| x + 1).ok();
+		match pruned_idx.or(next_idx) {
 			None => None,
 			Some(idx) => {
 				// skip by the number of elements pruned in the preceding subtrees,
 				// which is the sum of the size of each subtree
 				Some(
 					self.pruned_nodes[0..(idx as usize)]
 						.iter()
-						.map(|n| (1 << (bintree_postorder_height(*n) + 1)) - 1)
+						.map(|n| {
+							let height = bintree_postorder_height(*n);
+							if height == 0 {
+								// a leaf node can be pruned but will not be compacted
+								// so we do not shift for a leaf node
+								0
+							} else {
+								// height 1, 3 nodes, offset 2 = 1 + 1
+								// height 2, 7 nodes, offset 6 = 3 + 3
+								// height 3, 15 nodes, offset 14 = 7 + 7
+								2 * ((1 << height) - 1)
```

Comment:
I believe this returns zero for a height of zero, consider removing the `if`?

Reply:
Oh nice. That feels better than having that (redundant) special case... 😀